OpenRA.Mods.D2k/Traits/World/D2kResourceRenderer.cs

```diff
@@ -101,12 +100,13 @@ public enum ClearSides : byte
 			{ ClearSides.Bottom | ClearSides.TopLeft | ClearSides.BottomLeft | ClearSides.BottomRight, 49 },
 		};
 
-		public D2kResourceLayer(Actor self)
-			: base(self) { }
+		public D2kResourceRenderer(Actor self, D2kResourceRendererInfo info)
+			: base(self, info) { }
 
 		bool CellContains(CPos c, ResourceType t)
 		{
-			return RenderContent.Contains(c) && RenderContent[c].Type == t;
+			var type = ResourceLayer.GetResourceType(c);
+			return RenderContent.Contains(c) && type == t;
```

Comment:
(Very) Minor nit: If `!RenderContent.Contains(c)` we'd unnecessarily call `GetResourceType`, maybe `return RenderContent.Contains(c) && ResourceLayer.GetResourceType(c) == t;` is better.